modules/3d/src/pointcloud/io_ply.hpp

```diff
@@ -18,19 +18,37 @@ enum class DataFormat
     BinaryBigEndian
 };
 
+struct Property
+{
+    bool isList;
+    int counterType;
+    int valType;
+    std::string name;
+};
+
+struct ElementDescription
+{
+    size_t amount;
+    std::vector<Property> properties;
+};
+
 class PlyDecoder CV_FINAL : public BasePointCloudDecoder
 {
 public:
     void readData(std::vector<Point3f> &points, std::vector<Point3f> &normals, std::vector<Point3_<uchar>> &rgb, std::vector<std::vector<int32_t>> &indices) CV_OVERRIDE;
 
 protected:
     bool parseHeader(std::ifstream &file);
-    void parseBody(std::ifstream &file, std::vector<Point3f> &points, std::vector<Point3f> &normals, std::vector<Point3_<uchar>> &rgb);
+    void parseBody(std::ifstream &file, std::vector<Point3f> &points, std::vector<Point3f> &normals, std::vector<Point3_<uchar>> &rgb,
+                   std::vector<std::vector<int32_t>> &indices);
```

Comment:
why not just int?

Reply:
For consistency purposes
This data type is used everywhere across point cloud / mesh loading for indices